jira/src/test/java/org/arquillian/extension/governor/jira/impl/reporter/JiraGovernorRecorderTest.java

```diff
@@ -0,0 +1,160 @@
+package org.arquillian.extension.governor.jira.impl.reporter;
+
+import org.arquillian.extension.governor.api.detector.Detector;
+import org.arquillian.extension.governor.api.detector.utils.OS;
+import org.arquillian.extension.governor.jira.api.Jira;
+import org.arquillian.extension.governor.jira.configuration.JiraGovernorConfiguration;
+import org.arquillian.recorder.reporter.PropertyEntry;
+import org.arquillian.recorder.reporter.event.PropertyReportEvent;
+import org.arquillian.recorder.reporter.model.entry.KeyValueEntry;
+import org.arquillian.recorder.reporter.model.entry.table.TableCellEntry;
+import org.arquillian.recorder.reporter.model.entry.table.TableEntry;
+import org.arquillian.recorder.reporter.model.entry.table.TableRowEntry;
+import org.jboss.arquillian.core.api.Event;
+import org.jboss.arquillian.core.api.Instance;
+import org.jboss.arquillian.test.spi.event.suite.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class JiraGovernorRecorderTest {
+
+    private final static String JIRA_SERVER = "https://jboss.issues.org";
+
+    @Mock
+    private Event<PropertyReportEvent> propertyReportEvent;
+
+    @Mock
+    private Instance<JiraGovernorConfiguration> jiraGovernorConfigurationInstance;
+
+    @Captor
+    ArgumentCaptor<PropertyReportEvent> propertyReportEventArgumentCaptor;
+
+    private JiraGovernorRecorder jiraGovernorRecorder;
+
+    private JiraGovernorConfiguration jiraGovernorConfiguration;
+
+    @Before
+    public void setup() {
+        jiraGovernorConfiguration = new JiraGovernorConfiguration();
+        jiraGovernorConfiguration.setServer(JIRA_SERVER);
+        jiraGovernorRecorder = new JiraGovernorRecorder();
+        jiraGovernorRecorder.setPropertyReportEvent(propertyReportEvent);
+        jiraGovernorRecorder.setJiraGovernorConfigurationInstance(jiraGovernorConfigurationInstance);
+        when(jiraGovernorConfigurationInstance.get()).thenReturn(jiraGovernorConfiguration);
+    }
+
+    @Test
+    public void shouldReportJiraGovernorParamsForMethod() {
+
+        try {
+            After after = new After(FakeTestClass.class, FakeTestClass.class.getMethod("dummyTest"));
+              jiraGovernorRecorder.jiraReportEntries(after);
+        } catch (NoSuchMethodException e) {
+            e.printStackTrace();
```

Comment:
I agree, in real situation this branch will never be reached. but we are using getMethod which is throwing NoSuchMethodException, We should caught it or declare it to be thrown to avoid compilation problem.

In this case I should use throws instead of try catch.